lib/vector/Vlib/copy.c

```diff
@@ -641,10 +653,12 @@ int Vect_copy_tables(struct Map_info *In, struct Map_info *Out, int field)
                 _("Unable to copy table <%s> for layer %d from <%s> to <%s>"),
                 Fi->table, Fi->number, Vect_get_full_name(In),
                 Vect_get_name(Out));
+            Vect_destroy_map_struct(Out);
+            Vect_destroy_field_info(Fi);
             return -1;
         }
     }
-
+    Vect_destroy_field_info(Fi);
```

Comment:
(This is correct.)

Reply:
> (This is correct.)

My mistake, this must be in the above for-loop (at all places needed).

Reply:
Thanks it worked.

Reply:
Good, but there is another if statement in which Fi needs to be destroyed.
I would suggest to take another approach, instead of call Vect_destroy_field_info 3 times in such a short function, extract needed info from the Fi struct and afterwards immediately destroy it.

Add after the `if (Fi == NULL) {...}` statement something like:

```c
int fi_number = Fi->number;
const char * fi_name = Fi->name;
Vect_destroy_field_info(Fi);
```

All other Vect_destroy_field_info can be removed and use the new variables where needed.

Reply:
Ahh, never mind the above suggestion. Vect_destroy_field_info will free `Fi->name` too and `fi_name` will point to a freed string so that won't work. Please, just add the third Vect_destroy_field_info.

Reply:
Done

Reply:
Not quite :) No need to destroy what failed to be created. I was thinking of this if statement:

```c
if (field > 0 && Fi->number != field)
            continue;
```

Reply:
Done